tests/test_tensor/test_gpt2.py

```diff
@@ -50,33 +47,39 @@ def check_grad_equal(model, torch_model, pg: ProcessGroup):
 
 
 def run_gpt(init_spec_func, use_ddp):
-    set_seed(13234)
     world_size = torch.distributed.get_world_size()
+
+    # build a PG with TP and DP hybrid
     pg = ProcessGroup(dp_degree=(2 if (use_ddp and world_size >= 2) else 1))
+
+    # set seed make processes of the same tp group use the same seed
+    # set_seed(pg.tp_local_rank())
+
     get_components_func = non_distributed_component_funcs.get_callable('gpt2')
     model_builder, train_dataloader, test_dataloader, optimizer_class, criterion = get_components_func()
 
+    # make sure torch_model and model has the same parameter values
     with ColoInitContext(device=get_current_device()):
         model = model_builder()
     model = model.cuda()
     torch_model = model_builder().cuda()
-    if use_ddp:
-        # torch_model = DDP(torch_model, device_ids=[pg.rank()], process_group=pg)
-        # torch.distributed.barrier()
-        torch_model = DDP(torch_model,
-                          device_ids=[gpc.get_global_rank()],
-                          process_group=gpc.get_group(ParallelMode.DATA))
 
+    if use_ddp:
+        torch_model = DDP(torch_model, device_ids=[pg.rank()], process_group=pg.dp_process_group())
         model = ColoDDP(model, process_group=pg)
+
     for torch_p, p in zip(torch_model.parameters(), model.parameters()):
         torch_p.data.copy_(p)
 
     init_spec_func(model, pg)
+
     check_param_equal(model, torch_model, pg)
-    model.train()
-    torch_model.train()
-    torch.distributed.barrier()
 
+    # close the dropout in eval mode
+    model.eval()
+    torch_model.eval()
+
+    torch.distributed.barrier()
```

Comment:
set_seed(pg.tp_local_rank())
add it here

Reply:
done